hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/SstFilteringService.java

```diff
@@ -123,7 +133,7 @@ public BackgroundTaskResult call() throws Exception {
 
         long snapshotLimit = snapshotLimitPerTask;
 
-        while (iterator.hasNext() && snapshotLimit > 0) {
+        while (iterator.hasNext() && snapshotLimit > 0 && running.get()) {
```

Comment:
I don't think if `running.get()` check is really needed because [BackgroundService#shutdown()](https://github.com/apache/ozone/blob/a66e43e2838f087229cb4b0fac1e1dc969ab80dd/hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/utils/BackgroundService.java#L140) makes sure that all the active threads get terminated. Same for overriding `start()` and `shutdown()` functions.

Reply:
shutdown() will wait for the existing run method to run. It would be better if we break the loop & finish the run gracefully instead of killing the thread. Otherwise it could possibly lead to another set of problems.

Reply:
I don't agree with it.

Reply:
What do you propose here? I believe we dont have to run the whole loop for no reason.

Reply:
those services are not running on follower. As far as in understand we are saving on a many redundant operations.

Reply:
My point is you are not achieving much with this because `SSTFilteringService` will run after the reloading and do the clean-up again. You can argue that it is failing fast but I don't see much value of that here because it is background service and no client is waiting on it.

Anyways, if you believe this is the best and cleaner approach, please go ahead.

Reply:
yeah sure thanks.